cartoframes/context.py

```diff
@@ -175,20 +178,30 @@ def write(self, df, table_name, temp_dir='/tmp', overwrite=False,
                                                     geom_col, pgcolnames)
             self._set_schema(df, final_table_name, pgcolnames)
 
-        # create geometry column from lat/longs if requested
+        # create geometry column from long/lats if requested
         if lnglat:
-            # TODO: make this a batch job if it is a large dataframe or move
-            #       inside of _send_dataframe and/or batch
             tqdm.write('Creating geometry out of columns '
                        '`{lng}`/`{lat}`'.format(lng=lnglat[0],
                                                 lat=lnglat[1]))
-            self.sql_client.send('''
-                UPDATE "{table_name}"
-                SET the_geom = CDB_LatLng("{lat}"::numeric,
-                                          "{lng}"::numeric)
-            '''.format(table_name=final_table_name,
-                       lng=lnglat[0],
-                       lat=lnglat[1]))
+            query = '''
+                    UPDATE "{table_name}"
+                    SET the_geom = CDB_LatLng("{lat}"::numeric,
+                                              "{lng}"::numeric)
+                    '''.format(table_name=final_table_name,
+                               lng=lnglat[0],
+                               lat=lnglat[1])
+            if df.shape[0] > 100000:
+                batch_client = BatchSQLClient(self.auth_client)
+                status = batch_client.create([query, ])
+                tqdm.write('Table successfully written to CARTO: '
+                           '{base_url}dataset/{table_name} . `the_geom` '
```

Comment:
To make sure the `/` are there:
```
os.path.join(base_url, 'dataset', table_name)
```

Reply:
Maybe it's ok the way you have it, it's in other places in the code

Reply:
Might as well be careful